stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
+    // [time.zone.db]
+
+    // TRANSITION: work in progress
+    struct tzdb {
+        vector<leap_second> leap_seconds;
+        bool _All_ls_positive;
+    };
+
+    // TRANSITION: work in progress
+    class tzdb_list {
+    private:
+        using _ListType = forward_list<tzdb>;
+
+    public:
+        using const_iterator = _ListType::const_iterator;
+
+        tzdb_list(const tzdb_list&) = delete;
+        tzdb_list& operator=(const tzdb_list&) = delete;
+        tzdb_list()                            = default;
+
+        const tzdb& front() const noexcept {
+            _Shared_lock _Lk(_Tzdb_mutex);
+            return _Tzdb_list.front();
+        }
+
+        template <class... _ArgsTy>
+        void _Emplace_front(_ArgsTy&&... _Args) {
+            _Unique_lock _Lk(_Tzdb_mutex);
+            _Tzdb_list.emplace_front(_STD forward<_ArgsTy>(_Args)...);
+        }
+
+        const tzdb& _Reload() {
+            _Unique_lock _Lk(_Tzdb_mutex);
+            auto [_Leap_sec, _All_ls_positive] = _Xtzdb_generate_leap_seconds(_Tzdb_list.front().leap_seconds.size());
+            if (!_Leap_sec.empty()) {
+                _Emplace_front(tzdb{_STD move(_Leap_sec), _All_ls_positive});
+            }
+            return front();
+        }
+
+        void _Initialize() {
+            _Shared_lock _Shr_lk(_Tzdb_mutex);
+            if (_Tzdb_list.empty()) {
+                _Shr_lk._Unlock();
+                _Unique_lock _Lk(_Tzdb_mutex);
+                if (_Tzdb_list.empty()) {
+                    auto _Leap_sec = _Xtzdb_generate_leap_seconds(0);
+                    _Tzdb_list.emplace_front(tzdb{_STD move(_Leap_sec.first), _Leap_sec.second});
+                }
+            }
+        }
+
+    private:
+        _ListType _Tzdb_list;
+        mutable _Smtx_t _Tzdb_mutex = {};
+
+        struct _NODISCARD _Shared_lock {
+            explicit _Shared_lock(_Smtx_t& _Mtx_) : _Mtx{&_Mtx_} {
+                _Smtx_lock_shared(_Mtx);
+            }
+
+            void _Unlock() {
+                if (owns) {
+                    _Smtx_unlock_shared(_Mtx);
+                }
+                owns = false;
+            }
+
+            ~_Shared_lock() {
+                if (owns) {
+                    _Smtx_unlock_shared(_Mtx);
+                }
+            }
+
+            _Smtx_t* _Mtx;
+            bool owns = true;
+        };
+
+        struct _NODISCARD _Unique_lock {
+            explicit _Unique_lock(_Smtx_t& _Mtx_) : _Mtx{&_Mtx_} {
+                _Smtx_lock_exclusive(_Mtx);
+            }
+
+            ~_Unique_lock() {
+                _Smtx_unlock_exclusive(_Mtx);
+            }
+
+            _Smtx_t* _Mtx;
+        };
+    };
+
+    _INLINE_VAR tzdb_list _Global_tzdb_list;
+
+    inline tzdb_list& get_tzdb_list() {
+        _Global_tzdb_list._Initialize();
+        return _Global_tzdb_list;
+    }
+
+    inline const tzdb& get_tzdb() {
+        return _CHRONO get_tzdb_list().front();
+    }
+
+    inline const tzdb& reload_tzdb() {

```

Comment:
`_NODISCARD` here seems like a judgement call - technically, `reload_tzdb(); /* ... */ inspect(get_tzdb());` is valid, so I suppose it shouldn't be marked.

Reply:
As you say, it's not a pure observer, so my feeling was also to not mark it.